### PR TITLE
[Android] `list` operation

### PR DESCRIPTION
**Pull request by jackycodes**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

[AB#62461](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/62461)

Implemented `list` operation on React Native Android

- Data wrapper returns a `paginatedDocuments` object, which contains `hasNextPage()`, `getNextPage()` and `close()`.
- A UUID is used to uniquely identify each instance of a `list` operation
- Moved `getReadOptions()` and `getWriteOptions()` to utils

### CODE REVIEW

File: appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java [hunk 1]
```diff
@@ -117,48 +203,36 @@ public void accept(DocumentWrapper<T> documentWrapper) {
...
     }
 
-    private WriteOptions getWriteOptions(ReadableMap writeOptionsMap) {
-        WriteOptions writeOptions;
-        if (writeOptionsMap != null && writeOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
-            writeOptions = new WriteOptions(writeOptionsMap.getInt(TIME_TO_LIVE_KEY));
+    private static void addedDocumentToWritableArray(WritableArray writableArray, DocumentWrapper<JsonElement> documentWrapper) {
+        WritableMap jsDocumentWrapper = new WritableNativeMap();
+        addDocumentWrapperMetaData(documentWrapper, jsDocumentWrapper);
+
```
Comment (guperrot) [inline appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java]:
```suggestion
```

Comment (jackycodes) [reply appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java]: FIxed

File: appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java [hunk 2]
```diff
@@ -62,26 +77,104 @@ public String getName() {
...
+                Page<JsonElement> currentPage = documentWrappers.getCurrentPage();
+                if (currentPage.getError() != null) {
+                    DataException dataException = currentPage.getError();
+                    promise.reject("Failed list", dataException.getMessage(), dataException);
+                    return;
+                }
+                List<DocumentWrapper<JsonElement>> documents = currentPage.getItems();
+
+                /* Add documents to WritableArray */
+                for (DocumentWrapper<JsonElement> document : documents) {
```
Comment (guperrot) [inline appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java]: This crashes with null document

Comment (jackycodes) [reply appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java]: Fixed, please refer to commit: [358eb69c116b35bf9c4cde8acf46874965122bfa](https://github.com/microsoft/appcenter-sdk-react-native/pull/636/commits/358eb69c116b35bf9c4cde8acf46874965122bfa)

**Pull request merged**